src/middleware/passport-auth.ts

```diff
@@ -0,0 +1,168 @@
+            }
+        )
+    );
+
+    if (process.env.AUTH_PROVIDERS?.includes('onelogin')) {
+        const oneLoginIssuer = await Issuer.discover(`${process.env.ONELOGIN_URL}/.well-known/openid-configuration`);
+        const privateKey = process.env.ONELOGIN_PRIVATE_KEY!.replace(/\\n/g, '\n');
+
+        passport.use(
+            'onelogin',
+            new OpenIdStrategy(
+                {
+                    client: new oneLoginIssuer.Client(
+                        {
+                            client_id: process.env.ONELOGIN_CLIENT_ID!,
+                            client_secret: process.env.ONELOGIN_CLIENT_SECRET!,
+                            redirect_uris: [`${process.env.BACKEND_URL}/auth/onelogin/callback`],
+                            token_endpoint_auth_method: 'private_key_jwt',
+                            token_endpoint_auth_signing_alg: 'PS256',
+                            id_token_signed_response_alg: 'ES256'
+                        },
+                        {
+                            keys: [readPrivateKey(privateKey).export({ format: 'jwk' })]
+                        }
+                    ),
+                    params: {
+                        response_type: 'code',
+                        ui_locales: 'en',
+                        scope: 'openid email'
+                        // TODO: we need to update our OneLogin config to be able to request the below claims
+                        // vtr: ['P2.Cl.Cm'],
+                        // claims: {
+                        //     userinfo: {
+                        //         'https://vocab.account.gov.uk/v1/coreIdentityJWT': null
+                        //     }
+                        // }
+                    }
+                },
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                async (tokenset: TokenSet, userInfo: UserinfoResponse, cb: any): Promise<void> => {
+                    logger.debug('auth callback from onelogin received');
+
+                    if (userInfo.email === undefined) {
+                        logger.error('onelogin auth failed: account has no email address');
+                        cb(null, undefined, {
+                            message: 'onelogin account does not have an email, use another provider'
+                        });
+                        return;
+                    }
+
+                    try {
+                        logger.debug('checking if user has previously logged in...');
+                        const existingUser = await userRepository.findOneBy({ email: userInfo.email });
+
+                        if (existingUser && existingUser.provider !== 'onelogin') {
+                            logger.error('onelogin auth failed: email was registered via another provider');
+                            cb(null, undefined, { message: 'User is already registered via another provider' });
+                            return;
```

Comment:
We could potentially support multiple oauth providers per account, but that would require a more complex user setup so I've ignored it for now.

Reply:
done - SW-190